src/coreComponents/mesh/mpiCommunications/NeighborCommunicator.hpp

```diff
@@ -46,6 +46,37 @@ class NeighborCommunicator
   NeighborCommunicator():
     NeighborCommunicator( -1 ){};
 
+
+  enum class Location
+  {
+    Elem, //!< location is element (like pressure in finite volumes)
+    Face, //!< location is face (like flux in mixed finite elements)
+    Edge, //!< location is edge (like flux between fracture elements)
+    Node  //!< location is node (like displacements in finite elements)
+  };
+
+  struct fieldSyncKey
```

Comment:
```suggestion
  struct FieldSyncKey
```
Not a huge fan of the `Key` suffix, but we'll see in the end.

Reply:
for now it's only used as key which is why I added the key suffix but let's see what we end up doing. There is probably a better name as well.